sdk/identity/azure-identity/Troubleshoot.md

```diff
@@ -0,0 +1,223 @@
+
+##### Service principal with certificate
+
+| Variable name | Value |
+| --- | --- |
+AZURE_CLIENT_ID |ID of an Azure Active Directory application. |
+AZURE_TENANT_ID | ID of the application's Azure Active Directory tenant. |
+AZURE_CLIENT_CERTIFICATE_PATH | Path to a PEM-encoded certificate file including private key (without password protection). |
+
+##### Username and password
+
+| Variable name | Value |
+| --- | --- |
+AZURE_CLIENT_ID | ID of an Azure Active Directory application. |
+AZURE_USERNAME | A username (usually an email address). |
+AZURE_PASSWORD | The associated password for the given username. |
+
+### Client Authentication Error
+
+The `EnvironmentCredential` supports Service Principal authentication and Username + Password Authentication.
+Please follow the troubleshooting guidelines below for the respective authentication which you tried and failed.
+
+| Authentication Type | Troubleshoot Guide |
+| --- | --- |
+| Service Principal | [Service Principal Auth Troubleshooting Guide](#troubleshooting-username-password-authentication-issues) |
+| Username Password | [Username Password Auth Troubleshooting Guide](#troubleshooting-username-password-authentication-issues) |
+
+## Troubleshooting Username Password Authentication Issues
+
+### Two Factor Authentication Required Error
+
+The `UsernamePassword` credential works only for users whose two factor authentication has been disabled in Azure Active Directory. You can change the Multi Factor Authentication in Azure Portal by following the steps [here](https://docs.microsoft.com/azure/active-directory/authentication/howto-mfa-userstates#change-the-status-for-a-user).
+
+## Troubleshooting Service Principal Authentication Issues
+
+### Illegal/Invalid Argument Issues
+
+#### Client Id
+
+The Client Id is the application Id of the registered application / service principal in Azure Active Directory.
+It is a required parameter for `ClientSecretCredential` and `ClientCertificateCredential`. If you have already created your service principal
+then you can retrieve the client/app id by following the instructions [here](https://docs.microsoft.com/azure/active-directory/develop/howto-create-service-principal-portal#get-tenant-and-app-id-values-for-signing-in).
+
+#### Tenant Id
+
+The tenant id is te Global Unique Identifier (GUID) that identifies your organization. It is a required parameter for
+`ClientSecretCredential` and `ClientCertificateCredential`. If you have already created your service principal
+then you can retrieve the client/app id by following the instructions [here](https://docs.microsoft.com/azure/active-directory/develop/howto-create-service-principal-portal#get-tenant-and-app-id-values-for-signing-in).
+
+### Client Secret Credential Issues
+
+#### Client Secret Argument
+
+The client secret is the secret string that the application uses to prove its identity when requesting a token, this can also can be referred to as an application password.
```

Comment:
Super nitpicky:
```suggestion
The client secret is the secret string that the application uses to prove its identity when requesting a token; this can also can be referred to as an application password.
```